modules/engines/engine-javascript-graalium/src/main/java/org/eclipse/dirigible/engine/js/graalium/execution/platform/GraalJSFileSystem.java

```diff
@@ -0,0 +1,160 @@
+        if (isFollowLinks(linkOptions)) {
+            DELEGATE.checkAccess(path, modes.toArray(new AccessMode[0]));
+        } else if (modes.isEmpty()) {
+            DELEGATE.readAttributes(path, "isRegularFile", LinkOption.NOFOLLOW_LINKS);
+        } else {
+            throw new UnsupportedOperationException("CheckAccess for NIO Provider is unsupported with non empty AccessMode and NOFOLLOW_LINKS.");
+        }
+    }
+
+    @Override
+    public void createDirectory(Path dir, FileAttribute<?>... attrs) throws IOException {
+        DELEGATE.createDirectory(dir, attrs);
+    }
+
+    @Override
+    public void delete(Path path) throws IOException {
+        DELEGATE.delete(path);
+    }
+
+    @Override
+    public void copy(Path source, Path target, CopyOption... options) throws IOException {
+        DELEGATE.copy(source, target, options);
+    }
+
+    @Override
+    public void move(Path source, Path target, CopyOption... options) throws IOException {
+        DELEGATE.move(source, target, options);
+    }
+
+    @Override
+    public DirectoryStream<Path> newDirectoryStream(Path dir, DirectoryStream.Filter<? super Path> filter) throws IOException {
+        return DELEGATE.newDirectoryStream(dir, filter);
+    }
+
+    @Override
+    public void createLink(Path link, Path existing) throws IOException {
+        DELEGATE.createLink(link, existing);
+    }
+
+    @Override
+    public void createSymbolicLink(Path link, Path target, FileAttribute<?>... attrs) throws IOException {
+        DELEGATE.createSymbolicLink(link, target, attrs);
+    }
+
+    @Override
+    public Path readSymbolicLink(Path link) throws IOException {
+        return DELEGATE.readSymbolicLink(link);
+    }
+
+    @Override
+    public Map<String, Object> readAttributes(Path path, String attributes, LinkOption... options) throws IOException {
+        return DELEGATE.readAttributes(path, attributes, options);
+    }
+
+    @Override
+    public void setAttribute(Path path, String attribute, Object value, LinkOption... options) throws IOException {
+        DELEGATE.setAttribute(path, attribute, value, options);
+    }
+
+    @Override
+    public void setCurrentWorkingDirectory(Path currentWorkingDirectoryPath) {
+        this.currentWorkingDirectoryPath = currentWorkingDirectoryPath;
+    }
+
+    private static boolean isFollowLinks(final LinkOption... linkOptions) {
+        for (LinkOption lo : linkOptions) {
+            if (lo == LinkOption.NOFOLLOW_LINKS) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isSameFile(Path path1, Path path2, LinkOption... options) throws IOException {
+        return DELEGATE.isSameFile(path1, path2);
+    }
+
+    @Override
+    public Path getTempDirectory() {
+        return new File(System.getProperty("java.io.tmpdir")).toPath();
+    }
```

Comment:
Does this run well in a Docker container environment?

Reply:
Good question! I'll check on that